### DISCUSSION TITLE
How does this tool work?

### DISCUSSION THREAD

**Discussion by xantari** [valinet/ExplorerPatcher#763] [Q&A] [answered]
There's a section in the wiki that is supposed to detail how this tool works but it's empty right now.

From what I can tell this attaches itself to the explorer.exe process and does something with that process.

Does it look for specific memory positions in that process and intercept them?

Will this break with each new update microsoft pushes out because the memory locations could change?

Just trying to figure out if I should finally pull the trigger and update to windows 11 on my main PC and install this tool for a feature I know I will miss (Toolbars on the taskbar).  I updated to windows 11 on my other computers and seen what all the fuss is about, Microsoft really screwed up with Windows 11 with all the feature removals.

Just looking to get that one feature back that I use the most.

**Comment by valinet** [comment, marked as answer]
The section is empty because I have been busy working on other things. When I’ll have the time, I will update the article.

How does it work? Well, it’s  complicated. Yeah, it gets injected in explorer.exe and mainly patches various things. How does it do it? Via multiple strategies, like IAT patches, function patches based on offsets obtained from symbol files (https://github.com/valinet/ExplorerPatcher/wiki/Symbols), patches based on patterns in the file. Read more here: https://valinet.ro/2021/11/18/Functional-Windows-10-flyouts-on-Windows-11.html. Ofc it’s prone to breaking if the logic behind changes in a way that renders a patch (and its strategy) ineffective. I cannot predict the future. For the 22000-based builds, I think it won’t break to the point of being unfixable. For the next builds, idk, I will roll with them once the next version ships and see then.

**Comment by JohnBoy200** [comment]
I can absolutley reccomend this. I have lots of toolbars and was in dispair when I updated to W11 but this Explorer patch works really well and gives the same functionality and more as the W10 taskbar.
Valinet has/is doing a great job and with these latest version it couldn't be easier to keep it updated and I've never really had any problems.
So if you use toolbars and have W11 you definitely need this.

**Reply by valinet** [reply]
Thanks man, I appreciate it!

**Reply by xantari** [reply]
Yeah I’m glad this tool exists! But slightly dismayed that Microsoft necessitated this tools creation in the first place.
